src/main/flight/rpm_filter.c

```diff
@@ -45,162 +44,119 @@
 
 #include "rpm_filter.h"
 
-#define RPM_FILTER_MAXHARMONICS 3
-#define SECONDS_PER_MINUTE      60.0f
-#define ERPM_PER_LSB            100.0f
-#define MIN_UPDATE_T            0.001f
+#define RPM_FILTER_HARMONICS_MAX 3
+#define RPM_FILTER_DURATION_S    0.001f  // Maximum duration allowed to update all RPM notches once
+#define SECONDS_PER_MINUTE       60.0f
+#define ERPM_PER_LSB             100.0f
 
 
-static pt1Filter_t rpmFilters[MAX_SUPPORTED_MOTORS];
+typedef struct rpmFilter_s {
 
-typedef struct rpmNotchFilter_s {
+    int numHarmonics;
+    float minHz;
+    float maxHz;
+    float fadeRangeHz;
+    float q;
 
-    uint8_t  harmonics;
-    float    minHz;
-    float    maxHz;
-    float    fadeRangeHz;
-    float    q;
     timeUs_t looptimeUs;
+    biquadFilter_t notch[XYZ_AXIS_COUNT][MAX_SUPPORTED_MOTORS][RPM_FILTER_HARMONICS_MAX];
 
-    biquadFilter_t notch[XYZ_AXIS_COUNT][MAX_SUPPORTED_MOTORS][RPM_FILTER_MAXHARMONICS];
+} rpmFilter_t;
```

Comment:
Considering the 'rpm filter' is 'notch filter which center freq is determined by rpm', I think `rpmNotchFilter` is more readable

Reply:
The RPM Filter struct here is a collection of several variables and all RPM notches. Since we are calling this collection of variables "RPM Filter" whenever we talk about it, we immediately know what this struct is when we read the code.

The actual RPM notch filters are still called "notch" inside the RPM filter struct. The name "rpmNotchFilter_t" was a bit too specialized imo because the struct has more than a notch inside.

Reply:
ok then